util/src/main/java/io/zeebe/util/health/FailureListener.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.0. You may not use this file
+ * except in compliance with the Zeebe Community License 1.0.
+ */
+package io.zeebe.util.health;
+
+public interface FailureListener {
+  void onFailure();
+
+  void onRecovered();
+
+  static FailureListener withListeners(final Runnable onFailure, final Runnable onRecovered) {
```

Comment:
What is the advantage of creating an anonymous class from two runnables over a class implementing this interface?

```java
public class MySuperListener implements FailureListener {
  @Override
  public void onFailure() {
    // do stuff
  }
  
  @Override
  public void onRecovered() {
    // do other stuff
  }
}
```

If the issue is with name collisions, we can make the name more specific - call the interface `HealthMonitorableListener` and have the method be `onHealthMonitorableFailure()` and `onHealthMonitorableRecovery()`.